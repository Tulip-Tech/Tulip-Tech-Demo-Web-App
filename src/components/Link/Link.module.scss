@use 'src/styles/variables';
@use 'src/styles/theme';

.link {
  display: inline-block;
  color: var(--primary-color);
  font-weight: var(--body-font-weight-bold);
  text-decoration: none;

  &:hover,
  &:focus {
    text-decoration: underline;
  }
}
