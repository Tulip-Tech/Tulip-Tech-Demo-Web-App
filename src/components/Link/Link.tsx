import React, { MouseEventHandler } from 'react';
import { Link as RouterLink } from 'react-router-dom';
import classNames from 'classnames';

import styles from './Link.module.scss';

type Props = {
  className?: string;
  href?: string;
  to?: string;
  target?: string;
  children?: React.ReactNode;
  onClick?: MouseEventHandler<HTMLElement>;
};

const Link: React.FC<Props> = ({ to, href, children, className, onClick, target, ...rest }: Props) => {
  const linkClassName = classNames(styles.link, className);

  if (to) {
    return (
      <RouterLink to={to} className={linkClassName} onClick={onClick}>
        {children}
      </RouterLink>
    );
  }

  return (
    <a
      href={href}
      className={linkClassName}
      onClick={onClick}
      target={target}
      rel={
        // Security thingy for old browsers.
        // See https://mathiasbynens.github.io/rel-noopener/#recommendations
        target === '_blank' ? 'noreferrer' : undefined
      }
      {...rest}
    >
      {children}
    </a>
  );
};

export default Link;
