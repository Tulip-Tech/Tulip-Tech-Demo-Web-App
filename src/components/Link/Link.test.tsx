import React from 'react';
import { render } from '@testing-library/react';

import Link from './Link';

describe('<Link>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<Link />);

    expect(container).toMatchSnapshot();
  });
});
