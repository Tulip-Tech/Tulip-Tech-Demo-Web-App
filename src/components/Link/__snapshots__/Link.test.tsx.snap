// Vitest Snapshot v1

exports[`<Link> > renders and matches snapshot 1`] = `
<div>
  <a
    class="_link_a14c64"
  />
</div>
`;
